### PR TITLE
[corlib]: Use `Guid.NewGuid()` implementation from CoreFx.

### PR DESCRIPTION
**Pull request by baulig**
The corefx implementation of `Guid.NewGuid()` uses the internal `Interop.GetRandomBytes(byte*, int)`, which uses `RNGCryptoServiceProvider` directly without using `CryptoConfig`.

### CODE REVIEW

**Comment by baulig** [comment]
@monojenkins commit csproj

**Comment by baulig** [comment]
@monojenkins build failed

File: mcs/class/corlib/System/Guid.MonoTouch.cs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#if MONOTOUCH && FULL_AOT_RUNTIME
```
Comment (marek-safar) [inline mcs/class/corlib/System/Guid.MonoTouch.cs]: We should rename the file

Comment (baulig) [reply mcs/class/corlib/System/Guid.MonoTouch.cs]: How about `Guid.CommonCrypto.cs`?

Comment (baulig) [reply mcs/class/corlib/System/Guid.MonoTouch.cs]: I moved it into the `CommonCrypto/` directory and renamed it into `CommonCrypto/Guid.cs`.

**Pull request merged**